### PR TITLE
RFAI-04: Add revision endpoints and edit-before-approve flow

### PR DESCRIPTION
**Pull request by Chris0Jeky**
## Summary
- Add 3 proposal revision API endpoints (create/list/latest) as sub-resources on AutomationProposalsController, with EditorUserId derived from JWT claims
- Create proposalRevisionsApi.ts frontend HTTP client with typed interfaces
- Add edit-before-approve UI in PaperReviewView: Request edit opens an inline revision editor with reason tracking
- Store full sorted proposal operation snapshots for edits, including multi-operation proposals
- Harden revision editing against stale loads/saves, payload prop changes, string type coercion, invalid JSON, and apply/reject races while editing

## Scope note
- This PR stores and reviews revision payloads, but approval/execution still applies the original proposal operations. The remaining apply-revised-payload wiring is follow-up work before #976 can be closed.

## Test plan
- [x] 9 backend integration tests for revision endpoints, including invalid RevisedPayload JSON returning ValidationError
- [x] 5 frontend proposalRevisionsApi unit tests (create, list, latest, 404 handling, error rethrow)
- [x] frontend revision lifecycle/editor/view regressions: reparse on payload changes, preserve string field types, clear stale state on proposal switch, ignore stale saves, multi-op editable payload, and disable decisions while editing
- [x] dotnet test backend/tests/Taskdeck.Api.Tests/Taskdeck.Api.Tests.csproj -c Release --filter "FullyQualifiedName~ProposalRevisionApiTests"
- [x] npm run test -- --run src/tests/views/paper/review/ReviewRevisionEditor.spec.ts src/tests/composables/useProposalRevisions.spec.ts src/tests/views/paper/review/PaperReviewView.spec.ts
- [x] npm run typecheck

Refs #976

### CODE REVIEW

**Comment by Chris0Jeky** [comment]
## Adversarial Review Findings

### Fixed in `3c853929`
1. **Route parameter naming inconsistency** (LOW): Revision endpoints used `{proposalId}` while all other sub-resource endpoints use `{id}`. Fixed to `{id}` for consistency.
2. **Missing XML doc comments** (LOW): All existing endpoints had `<summary>` comments; the three revision endpoints did not. Added.

### Fixed in `0fe6bee0` (background agent)
3. **No guard against editing non-pending/expired proposals** (MED): `onRequestEdit` did not check proposal status before opening the editor. Fixed with status and expiry checks.
4. **No guard against multi-operation proposals** (MED): The editor only handles single-operation payloads but accepted multi-op proposals. Fixed with a toast guard.
5. **Silent error swallowing in loadRevisionState** (LOW): Catch block swallowed errors without feedback. Fixed to surface via toast.

### Fixed in next push
6. **Race condition in loadRevisionState** (P2, from Codex review): Rapid proposal switching could cause a slow response for a previous proposal to overwrite revision state for the current proposal. Fixed with a generation counter that discards stale responses. Test added.

### Assessed and acceptable
7. **editablePayload uses only ops[0].parameters** (P1, from Codex review): Valid for single-op proposals. Multi-op proposals are already blocked by the guard in `0fe6bee0`. When multi-op editing is supported in the future, `editablePayload` will need to serialize the full operations array.
8. **No MaxLength on CreateRevisionRequest.RevisedPayload** (N/A): The domain entity `ProposalRevision` validates emptiness and caps `Reason` at 500 chars. No payload size cap is consistent with how `Parameters` works on `ProposalOperation`.

File: frontend/taskdeck-web/src/views/paper/PaperReviewView.vue [hunk 1]
```diff
@@ -154,6 +156,26 @@ watch(
...
+} = useProposalRevisions(activeProposal)
+
+const editablePayload = computed(() => {
+  const p = activeProposal.value
+  if (!p) return '{}'
+  if (latestRevision.value) return latestRevision.value.revisedPayload
+  const ops = p.operations ?? []
+  if (ops.length === 0) return '{}'
+  const firstOp = ops[0]
+  return firstOp.parameters ?? '{}'
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/PaperReviewView.vue]: Addressed. Multi-op proposals are blocked at the UI level (commit 0fe6bee0). The P2 race condition in loadRevisionState was fixed with a generation counter (commit 7016a46f). When multi-op editing is supported, editablePayload will need to serialize the full operations array.

File: frontend/taskdeck-web/src/composables/useProposalRevisions.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  const editing = ref(false)
+  const saving = ref(false)
+  const revisionCount = ref(0)
+  const latestRevision = ref<ProposalRevision | null>(null)
+
+  async function loadRevisionState(proposalId: string) {
+    try {
+      const revisions = await proposalRevisionsApi.getRevisions(proposalId)
+      revisionCount.value = revisions.length
+      latestRevision.value =
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/composables/useProposalRevisions.ts:27]: Fixed in commit 7016a46f. Added a monotonic generation counter to loadRevisionState -- when the proposal ID changes, the generation increments, and any in-flight response with a stale generation is silently discarded. Test added to verify the behavior.

File: frontend/taskdeck-web/src/views/paper/PaperReviewView.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/PaperReviewView.vue]: Fixed in `0055f51e`. `editablePayload` now serializes the full sorted proposal operation snapshot instead of `ops[0].parameters`, and the PaperReviewView regression covers a multi-operation proposal.

File: frontend/taskdeck-web/src/composables/useProposalRevisions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/composables/useProposalRevisions.ts:27]: Fixed. The load path now checks both a generation token and the active proposal id before writing revision state, with a regression covering out-of-order loads.

File: frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+const reason = ref('')
+const parseError = ref(false)
+
+onMounted(() => {
+  try {
+    const parsed = JSON.parse(props.operationsPayload) as Record<string, unknown>
+    fields.value = Object.entries(parsed).map(([key, value]) => ({
+      key,
+      value: typeof value === 'string' ? value : JSON.stringify(value),
+    }))
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue]: Fixed in `0055f51e`. The editor now reparses via an immediate `watch` on `operationsPayload`; `ReviewRevisionEditor.spec.ts` verifies field updates after prop changes.

File: frontend/taskdeck-web/src/composables/useProposalRevisions.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  async function saveRevision(payload: CreateRevisionPayload) {
+    const proposal = activeProposal.value
+    if (!proposal) return
+
+    try {
+      saving.value = true
+      const revision = await proposalRevisionsApi.createRevision(proposal.id, payload)
+      latestRevision.value = revision
+      revisionCount.value += 1
+      editing.value = false
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/composables/useProposalRevisions.ts]: Fixed in `0055f51e`. Saves now use a separate generation token and active proposal id check, so stale save responses cannot write into the next proposal's revision state. Regression added.

File: frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue [hunk 5]
```diff
@@ -0,0 +1,169 @@
+<script setup lang="ts">
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue:1]: Fixed. `watch` is imported from Vue and `onMounted` is no longer used for one-shot parsing.

File: frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    fields.value = Object.entries(parsed).map(([key, value]) => ({
+      key,
+      value: typeof value === 'string' ? value : JSON.stringify(value),
+    }))
+    parseError.value = false
+  } catch {
+    fields.value = [{ key: 'payload', value: props.operationsPayload }]
+    parseError.value = true
+  }
+})
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue]: Fixed in `0055f51e`. Parsing is now handled by `watch(() => props.operationsPayload, parsePayload, { immediate: true })`, with a regression for payload changes after mount.

File: frontend/taskdeck-web/src/views/paper/PaperReviewView.vue [hunk 7]
```diff
@@ -154,6 +156,26 @@ watch(
...
+})
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/PaperReviewView.vue:188]: Fixed in `0055f51e`. The editable payload now includes the complete sorted operation list, not only the first operation's parameters. Added PaperReviewView coverage for this case.

File: frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+  if (!canSave.value) return
+
+  let revisedPayload: string
+  if (parseError.value) {
+    revisedPayload = fields.value[0]?.value ?? ''
+  } else {
+    const obj: Record<string, unknown> = {}
+    for (const field of fields.value) {
+      try {
+        obj[field.key] = JSON.parse(field.value) as unknown
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue:79]: Fixed in `0055f51e`. The editor records whether each original field was a string or JSON value and preserves string fields as strings on save. Regression covers a string that looks like JSON.

File: frontend/taskdeck-web/src/composables/useProposalRevisions.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+      toast.error(getErrorDisplay(e, 'Failed to load revision history').message)
+    }
+  }
+
+  watch(
+    () => activeProposal.value?.id,
+    (id) => {
+      editing.value = false
+      if (id) {
+        void loadRevisionState(id)
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply frontend/taskdeck-web/src/composables/useProposalRevisions.ts:49]: Fixed in `0055f51e`. Proposal switches now immediately clear editing/saving state plus `revisionCount`/`latestRevision` before starting the next load. Regression added.

**Comment by Chris0Jeky** [comment]
## Review Findings

### HIGH

- `frontend/taskdeck-web/src/views/paper/PaperReviewView.vue:490` / `backend/src/Taskdeck.Application/Services/AutomationExecutorService.cs:145` — Saved revisions are not applied when the user approves/executes the proposal. `onApply` still calls `handleApproveProposal(p.id)` or `handleExecuteProposal(p.id)`, and the executor still runs `proposal.Operations`, so the user can save an edit and then apply the original unedited operation set. This violates the linked issue #976 acceptance criterion: "Approval applies the latest saved revision, not stale original state." Expected fix: make approve/execute materialize the latest `ProposalRevision` into the operation set that policy validation, permission validation, execution, audit, and outcome recording use, or block approval while a latest revision exists until that compiler path is wired. Add a backend integration test that creates a proposal with operation payload A, saves revision payload B, approves/executes, and asserts the board mutation uses B and not A; add a frontend regression that applying a proposal with `latestRevision` follows the revised-payload path.

### MEDIUM

- `frontend/taskdeck-web/src/views/paper/review/ReviewRevisionEditor.vue:61` / `backend/src/Taskdeck.Application/Services/ProposalRevisionService.cs:34` — The editor can persist a broken operation snapshot while still passing backend validation. For fields originally parsed as JSON, `onSave` catches `JSON.parse` failures and stores the raw text as a string, so editing the `operations` textarea to invalid JSON emits a payload like `{ "operations": "not valid json" }`. The backend only checks that `RevisedPayload` is syntactically valid JSON, not that it remains a full operation snapshot. Expected fix: keep per-field JSON parse errors visible and block save for JSON fields until valid, and validate the revision payload shape server-side (`operations` array with operation entries) before accepting it. Add component coverage proving invalid operation JSON does not emit `save`, plus API/service coverage rejecting a syntactically valid but semantically invalid revision payload.

### Notes

Existing review threads were inspected. The prior stale-load/stale-save, prop reparse, string-preservation, and full-operation snapshot comments appear addressed in `0055f51e`. The remaining Codex comment about approve/execute using the original operations is still a real blocker for this PR’s stated edit-before-approve behavior and for #976.

**Comment by Chris0Jeky** [comment]
## Review follow-up for #1058

Fixed in `2993b066` (`Apply latest proposal revision on execute`).

Findings addressed:
- **HIGH: saved revisions were not used by apply/execute** — `AutomationExecutorService` now materializes the latest `ProposalRevision` into an effective proposal before execution. The revised operations are used for policy validation, permission validation, operation dispatch, audit recording, and capture conversion sync. Added API coverage proving a proposal with original operation payload A and latest revision payload B creates the card from B, not A.
- **MEDIUM: semantically broken revision payloads were accepted** — `ProposalRevisionService` now validates that `RevisedPayload` is a full operation snapshot with a non-empty `operations` array and structurally valid operation entries. The revision editor now keeps JSON-valued fields invalid when their textarea content is malformed, shows an inline error, disables save, and does not emit `save`.

Verification:
- `dotnet test backend/tests/Taskdeck.Api.Tests/Taskdeck.Api.Tests.csproj -c Release --filter "FullyQualifiedName~ProposalRevisionApiTests"` — passed, 11/11.
- `dotnet test backend/tests/Taskdeck.Application.Tests/Taskdeck.Application.Tests.csproj -c Release --filter "FullyQualifiedName~ProposalRevisionServiceTests"` — passed, 1/1.
- `npm run test -- --run src/tests/views/paper/review/ReviewRevisionEditor.spec.ts` — passed, 3/3.
- `npm run typecheck` — passed.
- `git diff --check` — passed; only existing Windows LF-to-CRLF normalization warnings were reported.

**Comment by Chris0Jeky** [comment]
## CI Recovery Follow-up

The backend unit failures after `2993b066` were caused by the existing `AutomationExecutorServiceTests` harness not wiring `IUnitOfWork.ProposalRevisions`. The new revision-aware execution path correctly consults that repository even when there are no saved revisions, but the mock returned `null`, causing `NullReferenceException` in normal no-revision execution paths.

Fixed in `01a6bec1`:
- Added `IProposalRevisionRepository` to the executor test harness.
- Wired `IUnitOfWork.ProposalRevisions` to return the mock repository so default no-revision behavior is represented correctly.

Verification run locally in `.worktrees/pr-1058`:
- `dotnet test backend/tests/Taskdeck.Application.Tests/Taskdeck.Application.Tests.csproj -c Release --filter "FullyQualifiedName~AutomationExecutorServiceTests|FullyQualifiedName~ProposalRevisionServiceTests"` -- passed, 15/15.
- `dotnet test backend/tests/Taskdeck.Application.Tests/Taskdeck.Application.Tests.csproj -c Release` -- passed, 3124/3124.
- `git diff --check` -- passed, with only the existing Windows LF-to-CRLF normalization warning.

Fresh CI is running on the pushed commit.

**Comment by Chris0Jeky** [comment]
## Third Adversarial Review (Final) — Round 3

**No new findings.** All prior findings from rounds 1 and 2 verified as addressed.

### Verification of fix commits

**`2993b066` (Apply latest proposal revision on execute):**
- `MaterializeEffectiveProposalAsync` correctly queries the latest revision via `IUnitOfWork.ProposalRevisions.GetLatestByProposalIdAsync` and replaces the operations list using record `with` syntax
- All downstream execution paths (policy validation, permission validation, operation dispatch, audit recording, capture conversion sync) consistently use `effectiveProposal`
- `ProposalRevisionPayload.TryParseOperations` validates the full schema: root must be a JSON object with a non-empty `operations` array, each operation must have integer `sequence`, required string fields (`actionType`, `targetType`, `parameters`, `idempotencyKey`), and `parameters` must itself be a valid JSON object
- Error path when revision payload is invalid at execution time is properly logged and returned as a structured failure

**`01a6bec1` (Update executor revision test harness):**
- Correctly wires `Mock<IProposalRevisionRepository>` into `IUnitOfWork.ProposalRevisions`
- Moq default `null` return for `GetLatestByProposalIdAsync` correctly represents the no-revision happy path, allowing all pre-existing executor tests to pass without modification

### Edge cases verified as handled

1. **No revision exists** — `MaterializeEffectiveProposalAsync` returns the original proposal unchanged
2. **Revision saved after approval** — `ProposalRevisionService.CreateRevisionAsync` rejects with InvalidOperation (status guard)
3. **Unsaved edits block approval** — Frontend `revisionBusy` computed blocks Apply/Reject and shows informative toast
4. **Race condition on proposal switch** — Generation counters discard stale load/save responses (unit tested)
5. **Invalid JSON in editor** — Frontend blocks save; backend validates on create; backend re-validates on execute
6. **Multi-operation proposals** — Editor serializes full operations array sorted by sequence; backend parser handles arrays of any valid length >= 1
7. **Frontend payload format matches backend expectations** — `editablePayload` produces `{"operations":[...]}` with all required fields; optional fields (`id`, `targetId`, `expectedVersion`) handled gracefully when null/missing

### CI Status

All required checks passing (Backend Unit, API Integration, Frontend Unit, E2E Smoke, Architecture, Container Images, Migration Validation, OpenAPI Guardrail, Docs Governance, Security scans).

**Verdict: Ready for merge consideration.** No outstanding findings at any severity.
